test/watcher-middleware-test.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+var expect = require('chai').expect;
+var RSVP = require('rsvp');
+var watcherMiddleware = require('./../lib/index').watcherMiddleware;
+var fixture = require('./helpers/fixture-path');
+var TestHTTPServer = require('./helpers/test-http-server');
+
+describe('watcher-middleware', function() {
+  describe('watcher resolves correctly', function() {
+    var server;
+
+    afterEach(function() {
+      server.stop();
+      server = null;
+    })
+
+    it('sets the request & response headers correctly', function(done) {
+      var watcher = RSVP.Promise.resolve({
+        'directory': fixture('basic-file')
+      });
+
+      var middleware = watcherMiddleware(watcher, {
+        autoIndex: false
+      });
+
+      var wrapperMiddleware = function(req, resp /*next*/) {
+        middleware(req, resp, function() {
+          // assert request headers
+          var broccoliHeader = req.headers['x-broccoli'];
+          expect(broccoliHeader).to.have.property('url');
+          expect(broccoliHeader).to.have.property('filename');
+          expect(broccoliHeader).to.have.property('outputPath');
+
+          // assert response headers
+          expect(resp.getHeader('Cache-Control')).to.be.equal('private, max-age=0, must-revalidate');
+          done();
+        })
+      };
+
+      server = new TestHTTPServer(wrapperMiddleware);
```

Comment:
The wrapperMiddleware creates a closure to allow doing assertions. I could otherwise just create a another middleware to do the assertions after serveAssetMiddleware is finished executing.